Sources/Kanna/libxmlHTMLNode.swift

```diff
@@ -119,6 +122,36 @@ internal final class libxmlHTMLNode: XMLElement {
         let val = xmlPreviousElementSibling(self.nodePtr)
         return self.node(from: val)
     }
+    
+    // Setters for dark mode
+    fileprivate var internalStyles: [String: String]?
+    var styles: [String : String]? {
+        get {
+            // Cached lookup for performance
+            if let i = internalStyles {
+                return i
+            }
+            
+            if let style = self["style"] {
+                let styleItems = style.split(separator: ";")
```

Comment:
There are weird edge cases that break this, specifically with regards to malformed styling. THe correct way to implement this is using a lexer parser.

This should be noted, and unit test this, including in malformed css test cases, such as:

`color: red;; background: blue;`
`color:: red;`
`font-family: "FontName;"; color: yellow`